crates/polars-plan/src/dsl/function_expr/cat.rs

```diff
@@ -48,3 +64,42 @@ fn get_categories(s: &Column) -> PolarsResult<Column> {
     let arr = rev_map.get_categories().clone().boxed();
     Series::try_from((ca.name().clone(), arr)).map(Column::from)
 }
+
+/// Apply a function to the categories of a categorical column and re-broadcast the result back to
+/// to the array.
+fn apply_to_cats<F, T>(s: &Column, mut op: F) -> PolarsResult<Column>
+where
+    F: FnMut(&StringChunked) -> ChunkedArray<T>,
+    ChunkedArray<T>: IntoSeries,
+    T: PolarsDataType,
+{
+    let ca = s.categorical()?;
+    let (categories, phys) = match &**ca.get_rev_map() {
```

Comment:
Can this move into its own function, that saves monomorphizaton bloat.

Reply:
Does this resolve it? https://github.com/pola-rs/polars/pull/20211/commits/2b9edc38714261cffa02363479c6df502d14232f

Reply:
Yes, great!